cloud/src/autogluon/cloud/predictor/cloud_predictor.py

```diff
@@ -292,71 +291,99 @@ def _prepare_data(self, data, filename, output_type="csv"):
+        upload_image_path = None
+        if image_dir_path is not None:
+            image_zip_filename = image_dir_path
+            assert os.path.isdir(image_dir_path), "Please provide a folder containing the images"
+            image_zip_filename = os.path.basename(os.path.normpath(image_dir_path))
+            logger.log(20, "Zipping images ...")
+            zipfolder(image_zip_filename, image_dir_path)
+            image_zip_filename += ".zip"
+            logger.log(20, "Uploading images ...")
+            upload_image_path = self.sagemaker_session.upload_data(
+                path=image_zip_filename,
+                bucket=bucket,
+                key_prefix=key_prefix,
+            )
+            logger.log(20, "Images uploaded successfully")
+        return upload_image_path
+
+    def _find_common_path_and_replace_image_column(self, data, image_column):
+        common_path = os.path.commonpath(data[image_column].tolist())
+        common_path_head = os.path.split(common_path)[0]  # we keep the base dir to match zipping behavior
+        data[image_column] = [os.path.relpath(path, common_path_head) for path in data[image_column]]
+
+        return data, common_path
 
     def _upload_fit_artifact(
         self,
         train_data,
         tune_data,
         config,
         serving_script,
-        images=None,
+        image_column=None,
     ):
         cloud_bucket, cloud_key_prefix = s3_path_to_bucket_prefix(self.cloud_output_path)
         util_key_prefix = cloud_key_prefix + "/utils"
-        train_input = train_data
-        if isinstance(train_data, pd.DataFrame) or not is_s3_url(train_data):
-            train_data = self._prepare_data(train_data, "train")
-            logger.log(20, "Uploading train data...")
-            train_input = self.sagemaker_session.upload_data(
-                path=train_data, bucket=cloud_bucket, key_prefix=util_key_prefix
+
+        common_train_data_path = None
+        common_tune_data_path = None
+        if image_column is not None:
+            # Find common path to zip and replace image column with relative path to be used in remote environment
+            if isinstance(train_data, str):
+                train_data = load_pd.load(train_data)
+            else:
+                train_data = copy.deepcopy(train_data)
+            if tune_data is not None:
+                if isinstance(tune_data, str):
+                    tune_data = load_pd.load(tune_data)
+                else:
+                    tune_data = copy.deepcopy(tune_data)
+            train_data, common_train_data_path = self._find_common_path_and_replace_image_column(
+                data=train_data, image_column=image_column
             )
-            logger.log(20, "Train data uploaded successfully")
+            if tune_data is not None:
+                tune_data, common_tune_data_path = self._find_common_path_and_replace_image_column(
+                    data=train_data, image_column=image_column
```

Comment:
*Bug*: Shouldn't this be `data=tune_data`?

Reply:
Yes, this is indeed incorrect. Thanks for catching that!